### PR TITLE
Make sure SIGTERM shutdown the controller manager and scheduler API server

### PR DESCRIPTION
**Pull request by mfojtik**
Right know the upstream does not call the `server.Shutdown()` when the controller manager receives SIGTERM. The SIGTERM `stopCh` is not even wired into controller manager and scheduler.

In that case, when the controller manager or scheduler are terminated, their ports will keep open until kernel timeout is hit (~60s++) and the port is freed. Since for these two components we use host network, this is causing the operator observing container restart in pod and ultimately setting failure condition to true.

### CODE REVIEW

File: test/util/server/server.go [hunk 1]
```diff
@@ -660,14 +662,15 @@ func startKubernetesControllers(masterConfig *configapi.MasterConfig, adminKubeC
...
+	stopCh := make(chan struct{})
 	for key, value := range cmdLineArgs {
 		for _, token := range value {
 			args = append(args, fmt.Sprintf("--%s=%v", key, token))
 		}
 	}
 
 	go func() {
-		cmd := kube_controller_manager.NewControllerManagerCommand()
+		cmd := kube_controller_manager.NewControllerManagerCommand(stopCh)
```
Comment (deads2k) [inline test/util/server/server.go]: nit `utilwait.NeverStop` so we can find this agai

**Comment by deads2k** [comment]
Wow, this is a dense pull.  I've had to go back to review other blockers.  That's interesting

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go [hunk 2]
```diff
@@ -119,36 +119,44 @@ func (s *SecureServingInfo) Serve(handler http.Handler, shutdownTimeout time.Dur
...
 	shutDownTimeout time.Duration,
 	stopCh <-chan struct{},
-) error {
+) (<-chan struct{}, error) {
 	if ln == nil {
-		return fmt.Errorf("listener must not be nil")
+		return nil, fmt.Errorf("listener must not be nil")
 	}
 
+	doneCh := make(chan struct{})
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go]: this should go

Comment (mfojtik) [reply vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/secure_serving.go]: lemme push here after getting e2e results to dig

**Comment by deads2k** [comment]
flake fix, green merging

**Pull request merged**